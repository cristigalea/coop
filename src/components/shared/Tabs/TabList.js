import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, css } from 'aphrodite';

import TabHandle from './TabHandle';

const TabList = ({tabs, selectedTabIndex = 0, handleTabChange}) => {
    const handles = tabs.map((tab, index) => {
        
        return (
            <TabHandle 
                key={`tab-${index}`} 
                onClick={() => handleTabChange(index)} 
                isActive={index === selectedTabIndex}
            >
                {tab}
            </TabHandle>
        );
    })

    return (
        <ul className={css(styles.list)}>
            {handles}
        </ul>
    );
};

TabList.propTypes = {
    tabs: PropTypes.arrayOf(PropTypes.string).isRequired,
    selectedTabIndex: PropTypes.number,
    handleTabChange: PropTypes.func.isRequired
}

export default TabList;

const baseColor = '#054f9b';
const tabHeight = 70;

const styles = StyleSheet.create({
    list: {
        listStyle: 'none',
        paddingLeft: 0,
        display: 'flex',
        minHeight: `${tabHeight}px`,
        border: `2px solid ${baseColor}`,
        borderRadius: `${tabHeight/2}px`,
        overflow: 'hidden'
    }
});