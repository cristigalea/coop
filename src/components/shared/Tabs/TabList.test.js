import React from 'react';
import {shallow} from 'enzyme';

import TabList from './TabList';
import { StyleSheetTestUtils} from 'aphrodite';

describe('<TabList />', () => {
    beforeEach(() => {
        StyleSheetTestUtils.suppressStyleInjection();
    });
      
    afterEach(() => {
        StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    });

    it('should match the snapshot', () => {
        const component = shallow(<TabList tabs={['Tab1', 'Tab2', 'Tab3']} selectedTabIndex={1} handleTabChange={()=>{}} />);
        expect(component).toMatchSnapshot();
    });
});