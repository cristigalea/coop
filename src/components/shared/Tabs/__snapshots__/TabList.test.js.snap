// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabList /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TabList
    handleTabChange={[Function]}
    selectedTabIndex={1}
    tabs={
      Array [
        "Tab1",
        "Tab2",
        "Tab3",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TabHandle
          isActive={false}
          onClick={[Function]}
        >
          Tab1
        </TabHandle>,
        <TabHandle
          isActive={true}
          onClick={[Function]}
        >
          Tab2
        </TabHandle>,
        <TabHandle
          isActive={false}
          onClick={[Function]}
        >
          Tab3
        </TabHandle>,
      ],
      "className": "list_2yirk3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "tab-0",
        "nodeType": "function",
        "props": Object {
          "children": "Tab1",
          "isActive": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Tab1",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "tab-1",
        "nodeType": "function",
        "props": Object {
          "children": "Tab2",
          "isActive": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Tab2",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "tab-2",
        "nodeType": "function",
        "props": Object {
          "children": "Tab3",
          "isActive": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Tab3",
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TabHandle
            isActive={false}
            onClick={[Function]}
          >
            Tab1
          </TabHandle>,
          <TabHandle
            isActive={true}
            onClick={[Function]}
          >
            Tab2
          </TabHandle>,
          <TabHandle
            isActive={false}
            onClick={[Function]}
          >
            Tab3
          </TabHandle>,
        ],
        "className": "list_2yirk3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "tab-0",
          "nodeType": "function",
          "props": Object {
            "children": "Tab1",
            "isActive": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Tab1",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "tab-1",
          "nodeType": "function",
          "props": Object {
            "children": "Tab2",
            "isActive": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Tab2",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "tab-2",
          "nodeType": "function",
          "props": Object {
            "children": "Tab3",
            "isActive": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Tab3",
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
